src/tool.rs

```diff
@@ -207,25 +214,30 @@ impl Tool {
                     &compiler_detect_output,
                 )?;
                 let stdout = String::from_utf8_lossy(&stdout);
-                guess_family_from_stdout(&stdout, path, cargo_output)
+                guess_family_from_stdout(&stdout, path, args, cargo_output)
             } else {
-                guess_family_from_stdout(&stdout, path, cargo_output)
+                guess_family_from_stdout(&stdout, path, args, cargo_output)
             }
         }
-        let detect_family = |path: &Path| -> Result<ToolFamily, Error> {
-            if let Some(family) = cached_compiler_family.read().unwrap().get(path) {
+        let detect_family = |path: &Path, args: &[String]| -> Result<ToolFamily, Error> {
+            let cache_key = [path]
+                .iter()
+                .map(|p| p.as_os_str().to_owned())
+                .chain(args.iter().map(|a| OsStr::new(a).to_owned()))
+                .collect();
+            if let Some(family) = cached_compiler_family.read().unwrap().get(&cache_key) {
                 return Ok(*family);
             }
 
-            let family = detect_family_inner(path, cargo_output, out_dir)?;
+            let family = { detect_family_inner(path, args, cargo_output, out_dir)? };
```

Comment:
I don't think this parenthese is necessary:

```suggestion
            let family = detect_family_inner(path, args, cargo_output, out_dir)?;
```

Reply:
D'oh, you're right. Interdiff link: https://github.com/rust-lang/cc-rs/compare/17c805a543f09e60f8c25296f322c9dd12c319bd..5d2908124311708917ceee5a59a4dfe01f35d5ff